<?xml version="1.0"?>
<launch>

  <arg name="mav_name"       default="firefly"/>
  <arg name="enable_logging" default="false" />
  <arg name="log_file"       default="$(arg mav_name)" />

  <arg name="file_obj"/>
  <arg name="file_texture"/>

  <group ns="$(arg mav_name)">

  	<node name="vrglasses_for_robots_ros" pkg="vrglasses_for_robots_ros" type="vrglasses_for_robots_ros" respawn="false" clear_params="true" output="screen">
	  	<remap from="odometry"         to="vi_sensor/ground_truth/odometry" />
	  	
  		<param name="framerate"       value="200.0" />
  		<param name="camera_frame_id" value="$(arg mav_name)/visensor/base_link" />
  		<param name="mesh_obj_file"   value="$(arg file_obj)" />
  		<param name="texture_file"    value="$(arg file_texture)" />
	  	<param name="shader_folder"   value="$(find vrglasses_for_robots)/shaders" />
		<param name="render_far"      value="400.0" /><!-- the range between near and far should be as small as possible, this affects the accuracy of the render depth map-->
		<param name="render_near"     value="0.1" />
  	</node>

  </group>
</launch>
